### PR TITLE
make `Debug` impl for `Term` simpler

### PR DESCRIPTION
**Pull request by WaffleLapkin**
meow

### CODE REVIEW

File: compiler/rustc_middle/src/ty/mod.rs [hunk 1]
```diff
@@ -536,14 +536,10 @@ unsafe impl<'tcx> Sync for Term<'tcx> where &'tcx (Ty<'tcx>, Const<'tcx>): Sync
...
-        let data = if let Some(ty) = self.ty() {
-            format!("Term::Ty({ty:?})")
-        } else if let Some(ct) = self.ct() {
-            format!("Term::Ct({ct:?})")
-        } else {
-            unreachable!()
-        };
-        f.write_str(&data)
+        match self.unpack() {
+            TermKind::Ty(ty) => write!("Term::Ty({ty:?})"),
```
Comment (aDotInTheVoid) [inline compiler/rustc_middle/src/ty/mod.rs:540]:
```suggestion
            TermKind::Ty(ty) => write!(f, "Term::Ty({ty:?})"),
```

**Comment by Nilstrieb** [comment]
fuck

**Comment by aDotInTheVoid** [comment]
r=me when CI's green

**Comment by Nilstrieb** [comment]
@bors=aDotInTheVoid,Nilstrieb,BoxyUwU 
image

**Comment by Nilstrieb** [comment]
:point_up:

**Comment by aDotInTheVoid** [comment]
i think bors knows something jyn

**Comment by jyn514** [comment]
>:(

**Comment by aDotInTheVoid** [comment]
image

**Comment by BoxyUwU** [comment]
Should we file a security incident about some random person trying to r+ rust PRs? Could be a potential attacker

**Comment by jyn514** [comment]
clearly the solution is to make this pr more gay
image

**Comment by aDotInTheVoid** [comment]
> Should we file a security incident about some random person trying to r+ rust PRs? Could be a potential attacker. (they are a self proclaimed "hacker")

with the state of bors these days I wouldn't be surprised if this worked

**Comment by Nilstrieb** [comment]
i am so sorry i messed up :((((

**Comment by WaffleLapkin** [comment]
It's okay 💜

**Pull request merged**